hwy/contrib/math/math-inl.h

```diff
@@ -989,6 +1008,257 @@ HWY_INLINE V Log(const D d, V x) {
+
+  /* Evaluate the second polynomial  (Pi/4 <= x <= 0) */
+  V y2 = MulAdd(sin_p0, z, sin_p1);
+  y2 = MulAdd(y2, z, sin_p2);
+  y2 = Mul(y2, z);
+  y2 = MulAdd(y2, x, x);
+
+  /* select the correct result from the two polynomials */
+  xmm1 = IfThenElse(poly_mask, y2, y);
+  xmm2 = IfThenElse(poly_mask, y, y2);
+    
+  /* update the sign */
+  s = Xor(xmm1, sign_bit_sin);
+  c = Xor(xmm2, sign_bit_cos);
+}
+
+// Sixth degree poly
+template <class D, class V>
+HWY_INLINE void SinCos6(D d, 
+  TFromD<D> dp1, TFromD<D> dp2, TFromD<D> dp3, 
+  V x, V& s, V& c) {
+  using T = TFromD<D>;
+  using TI = MakeSigned<T>;
+  using DI = Rebind<TI, D>;
+  const DI di;
+  using VI = decltype(Zero(di));
+  using M = Mask<D>;
+
+  static constexpr size_t bits = sizeof(TI) * 8;
+  // equivalent to 0x80000000 for int32 and 0x8000000000000000L for int64
+  static constexpr TI mask = TI(1) << (bits - 1);
+  const VI sign_mask = Set(di, mask);
+  const VI inv_sign_mask = Set(di, ~mask);
+  const VI ci_0 = Zero(di);
+  const VI ci_1 = Set(di, 1);
+  const VI ci_inv_1 = Set(di, ~1);
+  const VI ci_2 = Set(di, 2);
+  const VI ci_4 = Set(di, 4);
+  const V cos_p0 = Set(d, T(-1.13585365213876817300E-11));
+  const V cos_p1 = Set(d, T(2.08757008419747316778E-9));
+  const V cos_p2 = Set(d, T(-2.75573141792967388112E-7));
+  const V cos_p3 = Set(d, T(2.48015872888517045348E-5));
+  const V cos_p4 = Set(d, T(-1.38888888888730564116E-3));
+  const V cos_p5 = Set(d, T(4.16666666666665929218E-2));
+  const V sin_p0 = Set(d, T(1.58962301576546568060E-10));
+  const V sin_p1 = Set(d, T(-2.50507477628578072866E-8));
+  const V sin_p2 = Set(d, T(2.75573136213857245213E-6));
+  const V sin_p3 = Set(d, T(-1.98412698295895385996E-4));
+  const V sin_p4 = Set(d, T(8.33333333332211858878E-3));
+  const V sin_p5 = Set(d, T(-1.66666666666666307295E-1));
+  const V FOPI = Set(d, T(1.2732395447351626861510701069801148)); // 4 / M_PI
+  const V DP1 = Set(d, dp1);
+  const V DP2 = Set(d, dp2);
+  const V DP3 = Set(d, dp3);
+
+  V xmm1, xmm2, sign_bit_sin, y;
+  VI imm0, imm2, imm4;
+
+  sign_bit_sin = x;
+
+  /* take the absolute value */
+  x = And(x, BitCast(d, inv_sign_mask));
+
+  /* extract the sign bit (upper one) */
+  sign_bit_sin = And(sign_bit_sin, BitCast(d, sign_mask));
+
+  /* scale by 4/Pi */
+  y = Mul(x, FOPI);
+
+  /* store the integer part of y in imm2 */
+  imm2 = ConvertTo(di, y);
+
+  /* j=(j+1) & (~1) (see the cephes sources) */
+  imm2 = Add(imm2, ci_1);
+  imm2 = And(imm2, ci_inv_1);
+
+  y = ConvertTo(d, imm2); 
+  imm4 = imm2;
+
+  /* get the swap sign flag for the sine */
+  imm0 = And(imm2, ci_4);
+  imm0 = ShiftLeft<bits-3>(imm0);
+
+  V swap_sign_bit_sin = BitCast(d, imm0);
+
+  /* get the polynomial selection mask for the sine*/
+  imm2 = And(imm2, ci_2);
+  imm2 = IfThenElseZero(Eq(imm2, ci_0), Set(di, -1));
+      
+  // Cast integer 0000 FFFF (negative int) to mmask type. Is there a better way?
+  M poly_mask = RebindMask(d, Lt(imm2, Zero(di)));
```

Comment:
It seems we want a mask saying whether imm2 == 0. Why not just go to poly_mask = Eq(imm2, ci_0)? (also above)